tonic/functional/to_averaged_timesurface.py

```diff
@@ -31,116 +65,75 @@ def to_averaged_timesurface(
-        timesurface[radius, radius] = 1
-
-        if cells[r]:
-            local_memory = np.array(cells[r])
-
-            # find events ej such that tj is in [ti-temporal_window, ti)
-            context = local_memory[:, 0] >= event["t"] - temporal_window
-            context &= local_memory[:, 0] < event["t"]
-
-            # find events ej such that xj is in [xi-radius,xi+radius]
-            context &= local_memory[:, 1] <= x + radius
-            context &= local_memory[:, 1] >= x - radius
-
-            # find events ej such that yj is in [yi-radius,yi+radius]
-            context &= local_memory[:, 2] <= y + radius
-            context &= local_memory[:, 2] >= y - radius
-
-            # taking into consideration different polarities
-            context &= local_memory[:, 3] == event["p"]
-
-            # get the neighborhood of center event
-            neighborhood = local_memory[context]
-
-            if len(neighborhood) != 0:
-
-                # get unique coordinates
-                unique_coord = np.unique(neighborhood[:, 1:3], axis=0)
-                for i, coord in enumerate(unique_coord):
-
-                    # get timestamp of matching coordinates from cell
-                    match = neighborhood[:, 1] == coord[0]
-                    match &= neighborhood[:, 2] == coord[1]
-
-                    # scale x and y to find their position on timesurface
-                    scaled_x = int(coord[0] - x + radius)
-                    scaled_y = int(coord[1] - y + radius)
-
-                    # for each neighbor use some of decay of past events
-                    if decay == "lin":
-                        tmp_ts = (neighborhood[match, 0] - event["t"]) / (3 * tau) + 1
-                        tmp_ts[tmp_ts < 0] = 0
-                        timesurface[scaled_x, scaled_y] = np.sum(tmp_ts)
-                    elif decay == "exp":
-                        timesurface[scaled_x, scaled_y] = np.sum(
-                            np.exp((neighborhood[match, 0] - event["t"]) / tau)
-                        )
-
-        else:
-            # initialising cell with an empty list
-            cells[r] = []
-
-        # save event inside the cell
-        cells[r].append((event["t"], x, y, event["p"]))
-
-        # save into all_surfaces
-        all_surfaces[index, :, :, :] = timesurface
-
-    return all_surfaces
+        array of histograms (numpy.Array with shape (n_cells, n_pols, surface_size, surface_size))
+  """
+  
+  radius = surface_size // 2
+  assert surface_size > 0 and surface_size <= cell_size
+  assert "x" and "y" and "t" and "p" in events.dtype.names
+  assert decay=='lin' or decay=='exp'
+
+  if num_workers>1:
+    try:
+      from joblib import Parallel, delayed
+    except Exception as e:
+      print("Error: num_workers>1 needs joblib installed.")
+    use_joblib = True
+    ts_jobs, cells_jobs = max(1, round(num_workers/4)), max(1, round(num_workers*3/4))
```

Comment:
out of interest, why scale the num_workers parameter here at all?

Reply:
Hi! I kept this configuration because during my simulation I saw that it was better to favour the cells parallelization over the time surfaces (i.e.) events ones, so I allocated more resources on cells. But it was pure trial and error, no rigorous approach :)

Reply:
but the parallel computation of surfaces and parallel computation of histograms happens sequentially, right? But if you're saying that setting both ts_jobs and cells_jobs to num_workers makes it slower, then it's fine. This PR looks good otherwise!

Reply:
You are right! Each cell has a dependency on the time surface computation. Hence, it does not make sense to divide jobs. I modified the code accordingly!